src/io.cpp

```diff
@@ -358,35 +302,37 @@ PngPipelineInput::PngPipelineInput(cairo_surface_t *cairoSurface, Camera camera,
     image.height = cairo_image_surface_get_height(cairoSurface);
 }
 
-PipelineInputList PromptPngPipelineInput() {
+PipelineInputList GetPngPipelineInput(const PipelineOptions &values) {
     // I'm not sure why, but i can't get an initializer list to work here. Probably something to do
     // with copying unique ptrs
     PipelineInputList result;
     cairo_surface_t *cairoSurface = NULL;
-    std::string pngPath;
-
+    std::string pngPath = values.png;
+    
     while (cairoSurface == NULL || cairo_surface_status(cairoSurface) != CAIRO_STATUS_SUCCESS) {
-        cairoSurface = cairo_image_surface_create_from_png(Prompt<std::string>("PNG Path").c_str());
+         cairoSurface = cairo_image_surface_create_from_png(pngPath.c_str());
     }
     int xResolution = cairo_image_surface_get_width(cairoSurface);
     int yResolution = cairo_image_surface_get_height(cairoSurface);
-    result.push_back(std::unique_ptr<PipelineInput>(
-                         new PngPipelineInput(cairoSurface, PromptCameraPhysical(xResolution, yResolution), CatalogRead())));
+
+    if (!values.focalLength || !values.pixelSize) {
+        std::cerr << "Error: No focal length/pixel size given." << std::endl;
+        exit(1);
+    } 
+
+    float focalLength = values.focalLength;
+    float pixelSize = values.pixelSize;
+    float focalLengthPixels = focalLength * 1000 / pixelSize;
```

Comment:
As we discussed, should use either field of view or focal length, depending on which is provided. I recommend creating a function `float FocalLengthFromOptions(const PipelineOptions &)` which uses `values.pixelSize` and `values.focalLength` when provided, or uses `FovToFocalLength(values.fov)` otherwise. You can make the default `pixelSize = -1`, then it's easy to tell if it's been overridden.

Reply:
Done. The FovToFocalLength takes two arguments, I assume xResolution is the same as horizontal resolution?

Reply:
yep